clis/teliod/src/main.rs

```diff
@@ -0,0 +1,183 @@
+        })
+    })
+}
+
+/// Simple telio daemon for Linux and OpenWRT
+#[derive(Parser)]
+#[clap()]
+#[derive(Serialize, Deserialize)]
+enum Cmd {
+    #[clap(about = "Starts the daemon")]
+    Start { config_path: String },
+    #[clap(
+        about = "Forces daemon to add a log, added for testing to be, to be removed when the daemon will be more mature"
+    )]
+    HelloWorld { name: String },
+    #[clap(about = "Same as 'status' but in json string")]
+    Stop,
+}
+
+fn main() -> Result<()> {
+    match Cmd::parse() {
+        Cmd::Start { config_path } => {
+            if TeliodDaemon::is_running()? {
+                eprintln!("Teliod is already running, stop it by calling `teliod stop`");
+            } else {
+                let file = File::open(config_path)?;
+                let config: TeliodDaemonConfig = serde_json::from_reader(file)?;
+                start_daemon(config)?;
+            }
+        }
+        cmd => {
+            if TeliodDaemon::is_running()? {
+                let response = DaemonSocket::send_command(
+                    &TeliodDaemon::get_ipc_socket_path()?,
+                    &serde_json::to_string(&cmd)?,
+                )?;
+
+                if response.as_str() == "OK" {
+                    println!("Command executed successfully");
+                } else {
+                    eprintln!("Command failed");
+                }
+            } else {
+                eprintln!("Teliod daemon is not running");
+            }
+        }
+    };
+
+    Ok(())
+}
+
+/// Starts the Teliod daemon which will run actual Telio.
+///
+/// # Arguments
+///
+/// * `config` - Daemon config.
+fn start_daemon(config: TeliodDaemonConfig) -> Result<()> {
+    println!("Starting Teliod daemon...");
+    match TeliodDaemon::start() {
+        Ok(ProcessType::Api) => {
+            // Verifying that the server had been started, but we have no way of synchronizing
+            // the verification with the daemon, so using retries instead.
+            // The delay values had been chose experimentally to get a fast, but efficient start of the daemon.
+            // This delay mostly depends on how quickly Telio initializes itself from inside the daemon.
+            DaemonSocket::send_command_with_retries(
+                &TeliodDaemon::get_ipc_socket_path()?,
+                "hello_world",
+                &[
+                    Duration::from_millis(400),
+                    Duration::from_millis(100),
+                    Duration::from_millis(100),
+                    Duration::from_millis(100),
+                    Duration::from_millis(500),
+                ],
+            )?;
+
+            println!("Teliod daemon started");
+
+            Ok(())
+        }
+        Ok(ProcessType::Daemon) => daemon_event_loop(config),
+        Err(e) => Err(e),
+    }
+}
+
+fn daemon_event_loop(config: TeliodDaemonConfig) -> Result<()> {
+    // Create logger
+    let (non_blocking_writer, _tracing_worker_guard) =
```

Comment:
shouldn't the logger be initialized right at the beginning of main()?

Reply:
Nvm, I've missed the logic behind this.. Perhaps the API process can have its own logger instance?